### PR TITLE
[Php80] Skip union callable on ClassPropertyAssignToConstructorPromotionRector

### PR DESCRIPTION
**Pull request by samsonasik**
Closes #1100

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector.php [hunk 1]
```diff
@@ -184,6 +191,39 @@ private function shouldSkipParam(Param $param): bool
...
+        }
+
+        $types = $propertyType->getTypes();
+        foreach ($types as $type) {
+            if ($type instanceof CallableType) {
+                return true;
+            }
+        }
+
+        return false;
```
Comment (TomasVotruba) [inline rules/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector.php]: I think similar logic is in TypedProperty rule.
Could you extract it and reuse it?

Comment (samsonasik) [reply rules/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector.php]: I will try.

Comment (samsonasik) [reply rules/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector.php]: implemented, I added `PropertyAnalyzer` to check it https://github.com/rectorphp/rector-src/pull/1101/commits/7b813b738aa2d572caa0b3bf88a323a96ecc5298

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you 😊

**Pull request merged**